package-builder/flox-build.mk

```diff
@@ -377,25 +386,27 @@ define BUILD_template =
 	    fi; \
 	  fi; \
 	done
-	@$(_mv) -f $$@.new $$@
-
-  # Prepare temporary log file for capturing build output for inspection.
-  $(eval $(_pvarname)_logfile := $(shell $(_mktemp) --dry-run --suffix=-build-$(_pname).log))
+	$(_VV_) $(_mv) -f $$@.new $$@
 
   # Insert mode-specific template.
   $(call BUILD_$(_build_mode)_template)
 
   # Select the desired build mode as we declare the result symlink target.
   $(_result): $(_pname)_$(_build_mode)_build
-	@# Take this opportunity to fail the build if we spot fatal errors in the log.
-	@if $(_grep) -q "flox build failed (caching build dir)" $($(_pvarname)_logfile); then \
-	  echo "ERROR: flox build failed (see $($(_pvarname)_logfile))" 1>&2; \
+	@# Take this opportunity to fail the build if we spot fatal errors in the
+	@# build output. Recall that we force the Nix build to "succeed" in all
+	@# cases so that we can persist the buildCache, so when errors do happen
+	@# this is communicated by way of a $out that is 1) a file and 2) contains
+	@# the string "flox build failed (caching build dir)".
+	$(_VV_) if [ -f $(_result) ] && $(_grep) -q "flox build failed (caching build dir)" $(_result); then \
+	  echo "ERROR: flox build failed (see $(_result)-log)" 1>&2; \
```

Comment:
question nonblocking: do we want to print this message even when the log has already been printed? Do we always want to create a `result-log` symlink, even for successful builds?

Reply:
> question nonblocking: do we want to print this message even when the log has already been printed?

Yes I think so, because there can be Nix-related output between the build failure message and the conclusion of the build invocation that makes it harder for people to spot. I'd suggest we leave it in there for now and wait for UX feedback for things we could/should omit from the output going forward.

> Do we always want to create a result-log symlink, even for successful builds?

I think yes, much the same way we create the symlink for the buildCache output. Note that the link is handled as all other links, so removed with `flox build clean`, etc.